main.py

```diff
@@ -0,0 +1,42 @@
+import typer
+from src.db import Database
+
+app = typer.Typer()
+
+@app.command()
+def create_tables():
+    db = Database("finance_tracker.db")
```

Comment:
This is the `__init__` for the database

```python
    def __init__(self) -> None:
        self._connection = sqlite3.connect('finance_tracker.db')
```

You don't have to pass the db name in the init so the following will fix it

```python
db = Database()
```

If you want yours to work, you'll have to change the signature of the `__init__` like so:

```
    def __init__(self, database_name: str) -> None:
        self._connection = sqlite3.connect(database_name)
```